Src/DependencyCollector/Shared/Implementation/Operation/CacheBasedOperationHolder.cs

```diff
@@ -41,6 +41,11 @@ public bool Remove(long id)
             return this.memoryCache.Remove(id.ToString(CultureInfo.InvariantCulture)) != null;
         }
 
+        /// <summary>
+        /// Adds telemetry tuple to MemoryCache. DO NOT call it for the id that already exists in the cache.
+        /// This is a known Memory Cache race-condition issue when items with same id are added concurrently
+        /// and MemoryCache leaks memory. It should be fixed sometime AFTER .NET 4.7.1.
+        /// </summary>
```

Comment:
Can/should we do some very lightweight spinlock or similar around preventing this situation since we will likely have to support older framework versions for a while?